base/c.jl

```diff
@@ -404,9 +434,16 @@ Example using an external library:
 
 The string literal could also be used directly before the function
 name, if desired `"libglib-2.0".g_uri_escape_string(...`
+
+It's possible to declare the ccall as `gc_safe` by using the `gc_safe = true` option:
+    @ccall gc_safe=true strlen(s::Cstring)::Csize_t
```

Comment:
🚲 If `gc_safe=false` is the default, perhaps marking a `ccall` as `gc_safe` could just be spelled like:
```julia
@ccall :gc_safe strlen(s::Cstring)::Csize_t
```
(similar to how the `@atomic` and `@spawn` and `@assume_effects` macros take Symbols to control behaviour)

Reply:
We hope to be able to flip the default.

Reply:
How would that work (flipping the default)? What's the transition strategy?

Reply:
There isn't a firm strategy yet, right now anything using `cfunction`/`@ccallable` will transition correctly (this was done for foreign thread adoption) so the big issue is anything that uses the runtime. Already we have a trampoline in place and it may be feasible to add a transition check to that trampoline.

This would make calls into the runtime safe as well, but we wouldn't want to double transition for calls from Julia to the runtime, so we would need to manually mark those as `gc_safe=false`.

If the trampoline strategy works we should be able to flip the switch without any correctness issues, but it may be inefficient due to frequent transition and double transitions.